patches/ts-api-utils+0.0.21.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/ts-api-utils/package.json b/node_modules/ts-api-utils/package.json
+index 41ee37c..e4279d1 100644
+--- a/node_modules/ts-api-utils/package.json
++++ b/node_modules/ts-api-utils/package.json
+@@ -8,7 +8,7 @@
+ 	},
+ 	"license": "MIT",
+ 	"author": "Josh Goldberg <npm@joshuakgoldberg.com>",
+-	"type": "module",
++	"type": "commonjs",
```

Comment:
Should be fixed in https://github.com/JoshuaKGoldberg/ts-api-utils/pull/37 -> `ts-api-utils@0.0.22`.

Reply:
I guess the issue now is that TS won't let us import this file because it sees `type: 'module'` and our project is `type: 'commonjs'`

Reply:
Yup - we can't import the package without this patch.
TS looks at `type: module` and treats it as an exclusively esm package, erroring:
```
src/convert-comments.ts:1:24 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("ts-api-utils")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mts' or create a local package.json file with `{ "type": "module" }`.
       
1 import * as tools from 'ts-api-utils';
                          ~~~~~~~~~~~~~~
```

So we'll need to keep this patch (in perpetuity?)
Maybe this needs to be changed in the package so it doesn't list `type`?

Reply:
Hmm, I suppose we could take an issue. But I do wonder what the actual blockers to us being ESM are.

Reply:
> I do wonder what the actual blockers to typescript-eslint going ESM are.

In order for CJS to load ESM, it must use an `import()` expression.

ESLlint is CJS and does not use `import()` to load plugins - it uses `require` (via [`createRequire`](https://nodejs.org/api/module.html#modulecreaterequirefilename))
https://github.com/eslint/eslintrc/blob/main/lib/config-array-factory.js#L1086
https://github.com/eslint/eslintrc/blob/main/lib/config-array-factory.js#L984

So we need to export a CJS API from our packages. Now we could get around that and use ESM with a CJS facade in front of an ESM package and leverage `import()` expressions to load the ESM "backend", however our plugin and parser need to have fully synchronous APIs to adhere to the ESLint requirements.

Because the plugin needs to be CJS, all our lint utilities (`utils`, `type-utils`, `scope-manager`, etc) need to be CJS.
Because the parser needs to be CJS, all our parser utilities (`visitor-keys`, `typescript-estree`, etc) need to be CJS.

So for now we're firmly stuck as a project shipping CJS :(